readme.md

```diff
@@ -90,6 +96,15 @@ Treat uppercase and lowercase characters as being the same.
 
 Ensure you use this correctly. For example, files and directories should be matched case-insensitively, while most often, object keys should be matched case-sensitively.
 
+##### allPatterns
+
+Type: `boolean`\
+Default: `false`
+
+Requires every _`matcher()`_ pattern to have at least one positive match.
+
```

Comment:
Include some practical use-cases for this option. It might not be clear to the user.